File: resourcespoet/src/test/java/com/commit451/resourcespoet/ColorTest.java
package com.commit451.resourcespoet;

import org.junit.Assert;
import org.junit.Test;

import javax.xml.transform.stream.StreamResult;
import java.io.StringWriter;

/**
 * Tests the resources creation
 */
public class ColorTest {

    @Test
    public void colorTest() throws Exception {
        String text = Util.getFileText("color.xml");

        ResourcesPoet poet = ResourcesPoet.create()
                .addColor("color_primary", "#FF0000");

        StringWriter writer = new StringWriter();
        StreamResult result = new StreamResult(writer);
        poet.build(result, true);

        String writtenContent = writer.toString();
        System.out.println(writtenContent);

        Assert.assertEquals(Util.trimtrimtrim(text), Util.trimtrimtrim(writtenContent));
    }
}

File: src/test/kotlin/com/commit451/resourcespoet/ColorTest.kt
package com.commit451.resourcespoet

import org.junit.Test

/**
 * Tests the resources creation
 */
class ColorTest {

    @Test
    fun colorTest() {
        val poet = ResourcesPoet.create()
            .addColor("color_primary", "#FF0000")

        TestUtil.assertEquals("color.xml", poet)
    }
}
